File: frontend/src/api/order.ts
import request from './request'

export interface OrderItem {
    id?: number
    productId?: number
    productName: string
    productAttr?: string
    unit?: string
    quantity: number
    price: number
    costPrice?: number
    amount?: number
    costAmount?: number
    stockQuantity?: number
}

export interface Order {
    id?: number
    orderNo?: string
    type: string
    totalAmount?: number
    status?: number
    description: string
    createdBy?: string
    createdAt?: string
    customerId?: number
    supplierId?: number
    warehouseId?: number
    invoiceType?: string
    salesman?: string
    dealAmount?: number
    totalCost?: number
    orderDebt?: number
    discount?: number
    paymentStatus?: number
    customerAddress?: string
    customerPaymentInfo?: string
    items: OrderItem[]
    supplierName?: string
    customerName?: string
    warehouseName?: string
}

export interface PaymentRecord {
    id?: number
    paymentDate?: string
    paymentNo?: string
    sourceOrderNo?: string
    sourceOrderId?: number
    settlementMethod?: string
    amountDue?: number
    amountPaid?: number
    paymentPerson?: string
    payee?: string
    isFullyPaid?: boolean
    remark?: string
    type?: string // PAYMENT or RECEIPT
    payer?: string
    createdAt?: string
}

export const getOrders = (params: any) => request({ url: '/orders', method: 'get', params })
export const getOrder = (id: number) => request({ url: `/orders/${id}`, method: 'get' })
export const createOrder = (data: Order) => request({ url: '/orders', method: 'post', data })
export const updateOrder = (data: Order) => request({ url: '/orders', method: 'put', data })
export const deleteOrder = (id: number) => request({ url: `/orders/${id}`, method: 'delete' })
export const batchDeleteOrders = (ids: number[]) => request({ url: '/orders/batch-delete', method: 'post', data: ids })
export const updatePaymentStatus = (id: number, status: number) =>
    request({ url: `/orders/${id}/payment`, method: 'put', params: { status } })
export const updateInvoiceType = (id: number, invoiceType: string) =>
    request({ url: `/orders/${id}/invoice`, method: 'put', params: { invoiceType } })
export const getCustomerDebt = (params?: any) =>
    request({ url: '/orders/customer-debt', method: 'get', params })

// Payment records
export const getPayments = (params: any) => request({ url: '/orders/payments', method: 'get', params })
export const createPayment = (data: PaymentRecord) => request({ url: '/orders/payments', method: 'post', data })

// Receipt records
export const getReceipts = (params: any) => request({ url: '/orders/receipts', method: 'get', params })
export const createReceipt = (data: PaymentRecord) => request({ url: '/orders/receipts', method: 'post', data })

File: backend/order-service/src/main/java/com/example/orderservice/entity/Order.java
package com.example.orderservice.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("orders")
public class Order {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String orderNo;
    private Long customerId;
    private Long supplierId;
    private Long warehouseId;
    private String type; // PURCHASE, SALE
    private BigDecimal totalAmount;
    private Integer status; // 0-Pending, 1-Completed, 2-Cancelled
    private String description;
    private String createdBy;

    // 新增字段
    private String invoiceType;       // 票据类型
    private String salesman;          // 业务员
    private BigDecimal dealAmount;    // 成交金额
    private BigDecimal totalCost;     // 整单成本
    private BigDecimal orderDebt;     // 本单欠款
    private BigDecimal discount;      // 折扣百分比
    private Integer paymentStatus;    // 收/付款状态：0-未付, 1-部分, 2-已付
    private String customerAddress;   // 客户地址
    private String customerPaymentInfo; // 客户付款信息

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private List<OrderItem> items;

    @TableField(exist = false)
    private String supplierName;
    @TableField(exist = false)
    private String customerName;
    @TableField(exist = false)
    private String warehouseName;
}
